lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_release_backmerge_pull_request_action.rb

```diff
@@ -0,0 +1,207 @@
+            milestone: target_milestone&.number,
+            reviewers: reviewers,
+            team_reviewers: team_reviewers,
+            intermediate_branch_created_callback: intermediate_branch_created_callback
+          )
+        end.compact
+      end
+
+      # Determines the target branches for a release version.
+      #
+      # @param source_release_version [String] the source release version to compare against other release branches.
+      # @param default_branch [String] the default branch to use if no target branches are found.
+      # @return [Array<String>] the list of target branches greater than the release version.
+      def self.determine_target_branches(source_release_version:, default_branch:)
+        release_branches = Actions.sh('git', 'branch', '-r', '-l', 'origin/release/*').strip.split("\n")
+
+        all_release_branches_versions = release_branches
+                                        .map { |branch| branch.match(%r{origin/release/([0-9.]*)})&.captures&.first }
+                                        .compact
+
+        target_branches = all_release_branches_versions.select { |branch| Gem::Version.new(branch) > Gem::Version.new(source_release_version) }
+                                                       .map { |v| "release/#{v}" }
+        target_branches = [default_branch] if target_branches.empty?
+
+        target_branches
+      end
+
+      # Creates a backmerge pull request using the `create_pull_request` Fastlane Action.
+      #
+      # @param token [String] the GitHub token for authentication.
+      # @param repository [String] the repository where the pull request will be created.
+      # @param title [String] the title of the pull request.
+      # @param head_branch [String] the source branch for the pull request.
+      # @param base_branch [String] the target branch for the pull request.
+      # @param labels [Array<String>] the labels to add to the pull request.
+      # @param milestone [String] the milestone to associate with the pull request.
+      # @param reviewers [Array<String>] the individual reviewers for the pull request.
+      # @param team_reviewers [Array<String>] the team reviewers for the pull request.
+      # @return [void]
+      def self.create_backmerge_pr(token:, repository:, title:, head_branch:, base_branch:, labels:, milestone:, reviewers:, team_reviewers:, intermediate_branch_created_callback:)
+        commits_between_head_and_base = Fastlane::Helper::GitHelper.count_commits_between(base_ref: base_branch, head_ref: head_branch)
+
+        if commits_between_head_and_base.zero?
+          UI.message("No differences between #{head_branch} and #{base_branch}. Skipping PR creation.")
+          return nil
+        end
+
+        intermediate_branch = "merge/#{head_branch.gsub('/', '-')}-into-#{base_branch.gsub('/', '-')}"
+        Fastlane::Helper::GitHelper.create_branch(intermediate_branch)
+
+        intermediate_branch_created_callback&.call
```

Comment:
I'm not sure how this will be useful in practice if we don't provide it enough context, especially since this action can create multiple backmerge PRs (with different target branches), and will call the callback for each, and if we provide a callback we'll definitively want to do something slightly different for each one.

For example when in the case of Tumblr we'll want to use that to restore the version file to it's state on the branch targeted by the backmerge PR, we'll have to know what that branch is (`trunk`? or `release/N+1`?)

So I think we should at the very least pass the `base_branch` to the callback (and maybe optionally the name of the `intermediate_branch` that got created?)

Reply:
Indeed, makes sense. I guess most of the times `intermediate_branch` wouldn't have a lot of use; I can think of a case of having to go back to it if the callback checked out something else 🤔  I think I'll add it now as more context doesn't hurt. Update on 9977085a34a8ea6da056c762f48fa1a5d27e8ced